Secret-agency-service/src/test/java/cz/fi/muni/pa165/secretagency/service/facade/ReportFacadeTest.java

```diff
@@ -0,0 +1,226 @@
+package cz.fi.muni.pa165.secretagency.service.facade;
+
+import cz.fi.muni.pa165.secretagency.dto.ReportCreateDTO;
+import cz.fi.muni.pa165.secretagency.dto.ReportDTO;
+import cz.fi.muni.pa165.secretagency.dto.ReportUpdateTextDTO;
+import cz.fi.muni.pa165.secretagency.entity.Agent;
+import cz.fi.muni.pa165.secretagency.entity.Department;
+import cz.fi.muni.pa165.secretagency.entity.Mission;
+import cz.fi.muni.pa165.secretagency.entity.Report;
+import cz.fi.muni.pa165.secretagency.enums.*;
+import cz.fi.muni.pa165.secretagency.facade.ReportFacade;
+import cz.fi.muni.pa165.secretagency.service.config.ServiceConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceUnit;
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * Tests for report facade.
+ *
+ * @author Jan Pavlu
+ */
+@ContextConfiguration(classes = ServiceConfiguration.class)
+public class ReportFacadeTest extends AbstractTestNGSpringContextTests {
+
+    @PersistenceUnit
+    private EntityManagerFactory emf;
+
+    @Autowired
+    private ReportFacade reportFacade;
+
+    private Report transferBabisJrToKrymReport;
+
+    private Agent babis;
+
+    private Mission transferBabisJrToKrym;
+
+    @BeforeClass
+    public void setUp() {
+        EntityManager em1 = emf.createEntityManager();
+        em1.getTransaction().begin();
+
+        transferBabisJrToKrymReport = new Report();
+        transferBabisJrToKrymReport.setText("Get rid of my son");
+        transferBabisJrToKrymReport.setReportStatus(ReportStatus.NEW);
+        transferBabisJrToKrymReport.setDate(LocalDate.of(2018, 11, 1));
+        transferBabisJrToKrymReport.setMissionResult(MissionResultReportEnum.COMPLETED);
+
+        babis = new Agent();
+        babis.setLanguages(Collections.singleton(LanguageEnum.SK));
+        babis.setRank(AgentRankEnum.AGENT_IN_CHARGE);
+        babis.setName("Andrej Babis");
+        babis.setCodeName("Bures");
+        babis.setBirthDate(LocalDate.of(1980, 8, 8));
+
+        transferBabisJrToKrym = new Mission();
```

Comment:
:DDDDDD

Reply:
Udelame to s mockem teda?

Reply:
Pro to asi bych udělal vlastní třídu. Tyhle testy mi přijdou užitečný, jelikož už testují, že to celkově funguje.